src/sentry/static/sentry/app/views/organizationDiscover/result/index.jsx

```diff
@@ -43,8 +60,48 @@ export default class Result extends React.Component {
         view: 'table',
       });
     }
+
+    this.setState({
+      isEditMode: false,
+      savedQueryName: null,
+    });
   }
 
+  toggleEditMode = () => {
+    const {savedQuery} = this.props;
+    this.setState(state => {
+      const isEditMode = !state.isEditMode;
+      return {
+        isEditMode,
+        savedQueryName: isEditMode
+          ? savedQuery ? savedQuery.name : generateQueryName()
+          : null,
+      };
+    });
+  };
+
+  confirmSave = () => {
+    const {organization, queryBuilder} = this.props;
+    const {savedQueryName} = this.state;
+    const data = {...queryBuilder.getInternal(), name: savedQueryName};
+
+    createSavedQuery(organization, data)
+      .then(savedQuery => {
+        addSuccessMessage(t(`Successfully saved query ${savedQuery.name}`));
```

Comment:
Won't interpolating into the translated string make translations hard? Should you be using `tct()` instead.